forward/forward.go

```diff
@@ -52,40 +54,41 @@ func IdleTimeoutSetter(i time.Duration) optSetter {
 }
 
 func New(next http.Handler, setters ...optSetter) (*Forwarder, error) {
-	idleTimeoutPtr := new(time.Duration)
+	idleTimeout := 30 * time.Second
+
 	dialerFunc := func(network, addr string) (net.Conn, error) {
 		conn, err := net.DialTimeout(network, addr, time.Second*30)
 		if err != nil {
 			return nil, err
 		}
 
-		idleConn := idletiming.Conn(conn, *idleTimeoutPtr, func() {
+		idleConn := idletiming.Conn(conn, idleTimeout, func() {
 			if conn != nil {
 				conn.Close()
 			}
 		})
 		return idleConn, err
 	}
 
-	var timeoutTransport http.RoundTripper = &http.Transport{
+	timeoutTransport := &http.Transport{
 		Dial:                dialerFunc,
 		TLSHandshakeTimeout: 10 * time.Second,
+		MaxIdleTime:         idleTimeout / 2, // remove idle keep-alive connections to avoid leaking memory
 	}
+	log.Debugf("MaxIdleTime: %v", timeoutTransport.MaxIdleTime)
+	timeoutTransport.EnforceMaxIdleTime()
 	f := &Forwarder{
 		errHandler:   utils.DefaultHandler,
 		roundTripper: timeoutTransport,
 		next:         next,
-		idleTimeout:  30 * time.Second,
+		idleTimeout:  idleTimeout,
 	}
 	for _, s := range setters {
 		if err := s(f); err != nil {
 			return nil, err
 		}
 	}
 
-	// Make sure we update the timeout that dialer is going to use
-	*idleTimeoutPtr = f.idleTimeout
```

Comment:
Ah, thanks.  I've fixed this so that we use a pointer again.  FWIW, I've never actually seen an initialization pattern like ours used in Go.  The pattern I use in #75 is more typical (at least in how the standard library works).